src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -734,11 +733,32 @@ def all(self) -> None:
         -----
         Not a Number (NaN), positive infinity and negative infinity
         evaluate to True because these are not equal to zero.
+
+        Examples
+        --------
+        True, because nonzero integers are considered True.
+
+        >>> pd.Index([1, 2, 3]).all()
+        True
+
+        False, because 0 is considered False.
+
+        >>> pd.Index([0, 1, 2]).all()
+        False
         """
-        # TODO: SNOW-1458141 implement all
+        # If self is empty, the result for all is always True. This `self.empty` check results in one extra query.
+        # Passing an empty object through `query_compiler.all` results in the ValueError:
+        # The truth value of a Series is ambiguous. Use a.empty, a.bool(), a.item(), a.any() or a.all().
+        if self.empty:
```

Comment:
we can skip this check and just use to_series.all so we don't need this extra query right?

Reply:
`to_series.all()` does not return the correct value - instead it raises the ValueError I mentioned in the comment above

Reply:
It sounds like a bug. Does `pd.Series([]).all()` works?

Reply:
I got a `NotImplementedError: Snowpark pandas all API doesn't yet support non-integer/boolean columns` error

Reply:
Follow-up from in-person discussion - I fixed the bug when the Series is empty. Here's an old JIRA referring to it. https://snowflakecomputing.atlassian.net/browse/SNOW-1017231